In [49]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import tensorflow as tf

from sklearn import linear_model
from sklearn.preprocessing import OneHotEncoder

In [43]:
train_data = pd.read_csv("train_data.csv", delimiter = ";")
test_data = pd.read_csv("test_data.csv", delimiter = ";")

train_data.head()


#print(train_data.to_string())
x_train = train_data[["rel_1", "rel_2", "rel_3", "rel_4", "rel_5", "rel_6", "rel_7", "rel_8", "rel_9"]]
y_train = train_data[["wb_overall_mean"]]



#Rename column data to matching values
#train_data.iloc[1]
#y_test = test_data[["wb_overall_mean"]]


reg = linear_model.LinearRegression()
reg.fit(x_train,y_train)
print(reg.coef_)

#reg_pred = reg.predict(y_train)

[[ 0.204401   -0.06756281  0.04235165  0.05646504  0.03557858  0.01394905
  -0.06650978 -0.0040522   0.05508881]]


In [65]:
train_data[["denomination"]]

denomination
0     Christian (Roman Catholic)
1                            NaN
2                            NaN
3                            NaN
4     Christian (Roman Catholic)
...                          ...
9476                         NaN
9477                         NaN
9478                 Evangelical
9479                         NaN
9480              Muslim (Sunni)

[9481 rows x 1 columns]

In [ ]:
train_data.head()
#print(train_data.to_string())
#print(x_train)
ohe = OneHotEncoder(sparse_output=False)
#denomination_data = ["African religions, Buddhist, Christian, Christian (Roman Catholic), Christian (Protestant),Christian (Orthodox - Russian/Greek/etc.), Druze, Evangelical, Hindu, Jain, Jewish, Muslim, Muslim (Sunni), Muslim (non-sectarian), Muslim (Azhari), Muslim (Alevi), Shinto, Spiritist, Taoist, Other:"]
denomination_data = train_data[["denomination"]]
ohe.fit(denomination_data)

#encoded_data = ohe.transform(train_data)

print(denomination_data.shape)
#train_data


(9481, 1)


In [ ]:
train_data_dummies = pd.get_dummies(train_data)



Index(['subject', 'rel_1', 'rel_2', 'rel_3', 'rel_4', 'rel_5', 'rel_6',
       'rel_7', 'rel_8', 'rel_9',
       ...
       'denomination_Taoist', 'sample_type_general public',
       'sample_type_mixed', 'sample_type_online panel', 'sample_type_students',
       'compensation_course credit', 'compensation_monetary reward',
       'compensation_no compensation',
       'compensation_no compensation + course credit', 'compensation_raffle'],
      dtype='object', length=113)

In [76]:
list(train_data_dummies.columns)

['subject',
 'rel_1',
 'rel_2',
 'rel_3',
 'rel_4',
 'rel_5',
 'rel_6',
 'rel_7',
 'rel_8',
 'rel_9',
 'cnorm_1',
 'cnorm_2',
 'wb_gen_1',
 'wb_gen_2',
 'wb_phys_1',
 'wb_phys_2',
 'wb_phys_3',
 'wb_phys_4',
 'wb_phys_5',
 'wb_phys_6',
 'wb_phys_7',
 'wb_psych_1',
 'wb_psych_2',
 'wb_psych_3',
 'wb_psych_4',
 'wb_psych_5',
 'wb_psych_6',
 'wb_soc_1',
 'wb_soc_2',
 'wb_soc_3',
 'wb_overall_mean',
 'wb_phys_mean',
 'wb_psych_mean',
 'wb_soc_mean',
 'age',
 'ses',
 'education',
 'gdp',
 'gdp_scaled',
 'attention_check',
 'country_Australia',
 'country_Belgium',
 'country_Brazil',
 'country_Canada',
 'country_Chile',
 'country_China',
 'country_Croatia',
 'country_Denmark',
 'country_France',
 'country_Germany',
 'country_India',
 'country_Ireland',
 'country_Israel',
 'country_Italy',
 'country_Japan',
 'country_Lithuania',
 'country_Morocco',
 'country_Netherlands',
 'country_Romania',
 'country_Singapore',
 'country_Spain',
 'country_Turkey',
 'country_UK',
 'country_US',
 'gender_man',

In [34]:
print(test_data.to_string())

      subject      country     rel_1     rel_2  rel_3  rel_4     rel_5     rel_6     rel_7  rel_8  rel_9  cnorm_1  cnorm_2   age gender   ses  education               ethnicity                               denomination           gdp  gdp_scaled     sample_type                     compensation  attention_check
0        4079      Germany  0.000000  0.000000    0.0      0  0.166667  0.166667  0.166667   0.00   0.00     0.25     0.50  24.0  woman   7.0          5      Caucasian/European                                        NaN  48195.579904    0.590207           mixed                           raffle                1
1        1301       Canada  0.000000  0.142857    0.0      0  0.500000  0.166667  0.166667   0.00   0.25     0.25     0.25  49.0  woman   6.0          4      Caucasian/European                                        NaN  46210.547623    0.500519    online panel                  monetary reward                1
2        2689      Denmark  0.833333  0.714286    1.0      1  0.

Progress log: Read up on the data (conceptual analysis), read in the data into a jupyter notebook and fit a normal linear regression line. The regression takes the responses from the rel_ responses and compares it to the overall well-being score. Maybe what would be good to do is compare the coefficients to the other mean scores instead of just the overall. Maybe it would be good to run a correlation test. Regression coefficients:  [0.204401,-0.06756281,0.04235165,0.05646504,0.03557858,0.01394905,-0.06650978,-0.0040522,0.05508881]

In [10]:
test_data.head()

subject    country     rel_1     rel_2  rel_3  rel_4     rel_5     rel_6  \
0     4079    Germany  0.000000  0.000000    0.0      0  0.166667  0.166667   
1     1301     Canada  0.000000  0.142857    0.0      0  0.500000  0.166667   
2     2689    Denmark  0.833333  0.714286    1.0      1  0.666667  1.000000   
3     5909     Israel  0.000000  0.000000    0.0      0  0.166667  0.000000   
4       51  Australia  1.000000  1.000000    1.0      1  0.833333  1.000000   

      rel_7  rel_8  ...  gender  ses  education           ethnicity  \
0  0.166667   0.00  ...   woman  7.0          5  Caucasian/European   
1  0.166667   0.00  ...   woman  6.0          4  Caucasian/European   
2  0.666667   0.50  ...     man  6.0          4             African   
3  0.000000   0.00  ...     man  7.0          4       Mixed / other   
4  1.000000   0.75  ...     man  9.0          7  Caucasian/European   

  denomination           gdp  gdp_scaled   sample_type     compensation  \
0          NaN  48195.579904    0.590207         mixed           raffle   
1          NaN  46210.547623    0.500519  online panel  monetary reward   
2       Muslim  60726.466535    1.156377         mixed           raffle   
3          NaN  41613.998082    0.292837      students    course credit   
4       Muslim  57305.299016    1.001802  online panel  monetary reward   

   attention_check  
0                1  
1                1  
2                1  
3                1  
4                1  

[5 rows x 24 columns]

In [46]:
train_data.isna().sum()

subject               0
country               0
rel_1                 0
rel_2                 0
rel_3                 0
rel_4                 0
rel_5                 0
rel_6                 0
rel_7                 0
rel_8                 0
rel_9                 0
cnorm_1               0
cnorm_2               0
wb_gen_1              0
wb_gen_2              0
wb_phys_1             0
wb_phys_2             0
wb_phys_3             0
wb_phys_4             0
wb_phys_5             0
wb_phys_6             0
wb_phys_7             0
wb_psych_1            0
wb_psych_2            0
wb_psych_3            0
wb_psych_4            0
wb_psych_5            0
wb_psych_6            0
wb_soc_1              0
wb_soc_2              0
wb_soc_3            885
wb_overall_mean       0
wb_phys_mean          0
wb_psych_mean         0
wb_soc_mean           0
age                  25
gender                0
ses                   4
education             0
ethnicity           368
denomination       5118
gdp             